File: java-example/src/test/java/org/onflow/examples/java/verifySignature/userSignature/UserSignatureExampleTest.java
package org.onflow.examples.java.verifySignature.userSignature;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestAccount;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.common.test.TestAccount;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.cadence.BooleanField;
import org.onflow.flow.sdk.cadence.Field;

import static org.junit.jupiter.api.Assertions.*;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class UserSignatureExampleTest {

    @FlowTestAccount
    private TestAccount testAccount;

    @FlowTestAccount
    private TestAccount testAccount2;

    @FlowTestClient
    private FlowAccessApi accessAPI;

    private UserSignatureExample connector;

    @BeforeEach
    public void setup() {
        connector = new UserSignatureExample(accessAPI);
    }

    @Test
    public void canVerifyUserSignature() {
        try {
            Field<?> txResult = connector.verifyUserSignature(testAccount.getFlowAddress(), testAccount2.getFlowAddress());

            if (txResult instanceof BooleanField) {
                assertEquals(Boolean.TRUE, ((BooleanField) txResult).getValue(), "Signature verification failed");
            } else {
                fail("Expected BooleanField but got " + txResult.getClass().getSimpleName());
            }
        } catch (Exception e) {
            fail("Test failed with exception: " + e.getMessage());
        }
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/verifySiganture/userSignature/UserSignatureExampleTest.kt
package org.onflow.examples.kotlin.verifySiganture.userSignature

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Assertions.fail
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.examples.kotlin.verifySignature.userSignature.UserSignatureExample
import org.onflow.flow.common.test.*
import org.onflow.flow.sdk.*
import org.onflow.flow.sdk.cadence.BooleanField

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class UserSignatureExampleTest {
    @FlowTestAccount
    lateinit var testAccount: TestAccount

    @FlowTestAccount
    lateinit var testAccount2: TestAccount

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var connector: UserSignatureExample

    @BeforeEach
    fun setup() {
        connector = UserSignatureExample(accessAPI)
    }

    @Test
    fun `Can verify user signature`() {
        val txResult = connector.verifyUserSignature(testAccount.flowAddress, testAccount2.flowAddress)

        if (txResult is BooleanField) {
            assertTrue(txResult.value!!, "Signature verification failed")
        } else {
            fail("Expected BooleanField but got ${txResult::class.simpleName}")
        }
    }

}
